saleor/graphql/checkout/mutations.py

```diff
@@ -835,6 +836,8 @@ def perform_mutation(cls, _root, info, checkout_id, store_source, **data):
                 tracking_code=tracking_code,
                 redirect_url=data.get("redirect_url"),
             )
+        if info.context.site.settings.automatically_confirm_all_new_orders and order:
```

Comment:
there is an action `order_created` which is called when the order is created. IMO this part should be there